datafusion-examples/examples/to_char.rs

```diff
@@ -125,14 +125,14 @@ async fn main() -> Result<()> {
 
     assert_batches_eq!(
         &[
-            "+------------+",
-            "| t.values   |",
-            "+------------+",
-            "| 2020-09-01 |",
-            "| 2020-09-02 |",
-            "| 2020-09-03 |",
-            "| 2020-09-04 |",
-            "+------------+",
+            "+-----------------------------------+",
+            "| arrow_cast(t.values,Utf8(\"Utf8\")) |",
```

Comment:
I think this is a direct consequence of the fact that the original arrow-cast function was "rewritten" (well really planned) to a `CAST` which gets various special treatment throughout the codebase

Reply:
I think after @jayzhan211  has completed https://github.com/apache/arrow-datafusion/pull/9304 we will be able to avoid these changes by implementing `ArrowCastFunc::simplify`